### PR TITLE
modular rearchitecture attempt

### PR DESCRIPTION
**Pull request by rvion**

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -18,26 +16,12 @@ main = do
...
-    setImports ["Prelude", "System.IO"]
-    setTopLevelModules["Script"]
-    interpret "onMessage" (as :: Irc.MessageCallback)
-  case result of
-    Right callback -> return callback
-    Left error -> do
-      putStrLn $ "error interpreting: " ++ scriptFile
-      print error
-      exitFailure
+  (stdin',stdout') <- undefined -- execVe, returning a handle to stdin of the program and stdout
```
Comment (Globidev) [inline app/Main.hs]: So... 2 pipes then?
I don't know if there are ways of doing this easily, do you?

**Comment by rvion** [comment]
I just did, but I don't know how to sync from github desktop

**Comment by rvion** [comment]
ah, here is it :)

File: app/Main.hs [hunk 2]
```diff
@@ -18,26 +16,14 @@ main = do
...
-    setImports ["Prelude", "System.IO"]
-    setTopLevelModules["Script"]
-    interpret "onMessage" (as :: Irc.MessageCallback)
-  case result of
-    Right callback -> return callback
-    Left error -> do
-      putStrLn $ "error interpreting: " ++ scriptFile
-      print error
-      exitFailure
+  (Just hin, Just hout, _, _) <- createProcess (proc "ls" []){ std_out = CreatePipe }
```
Comment (Globidev) [inline app/Main.hs]: nice, wasn't familiar with this syntax

**Comment by Globidev** [comment]
looks great :+1:

**Comment by rvion** [comment]
ping @Globidev

**Pull request merged**